src/auth/auth.proto

```diff
@@ -157,26 +113,56 @@ message WhoAmIRequest {}
 
 message WhoAmIResponse {
   string username = 1;
-  bool is_admin = 2;
   int64 ttl = 3 [(gogoproto.customname) = "TTL"];
-  ClusterRoles cluster_roles = 4;
 }
 
 //// Authorization data structures
 
-// Scope (actually a "role" in canonical security nomenclature) represents a
-// rough level of access that a principal has to a repo
-enum Scope {
-  // To remove a user's scope from a repo, set their scope to NONE
-  NONE = 0;
-  READER = 1;
-  WRITER = 2;
-  OWNER = 3;
+// Roles represents the set of roles a principal has
+message Roles {
+  map<string, bool> roles = 1;
 }
 
-message ACL {
-  // principal -> scope. All principal names include the structured prefix indicating their type.
-  map<string, Scope> entries = 1;
+// RoleBinding represents the set of roles principals have on a given Resource
+message RoleBinding {
+  // principal -> roles. All principal names include the structured prefix indicating their type.
+  map<string, Roles> entries = 1;
+}
+
+// Permission represents the ability to perform a given operation on a Resource 
+enum Permission {
+  UNKNOWN = 0;
+
+  // CLUSTER_ADMIN encompasses all "admin" APIs, which are generally cluster configuration
+  CLUSTER_ADMIN        = 100;
```

Comment:
To keep things simple there's currently one "admin" permission in the admin role, but in a follow-on PR it makes sense to split this into a permission per admin RPC. I didn't tackle it here because it also involves rewriting the auth interceptor.

Reply:
This makes sense. Just checking my understanding of one piece of all this: this PR gets rid of the `FS` cluster role—is that replaced with binding a user to the `repoOwner` role, but at the cluster level?